mlrun/model_monitoring/db/stores/sqldb/sql_store.py

```diff
@@ -61,6 +61,7 @@ def __init__(
 
         self._sql_connection_string = kwargs.get("store_connection_string")
         self._engine = get_engine(dsn=self._sql_connection_string)
+        self._create_tables_if_not_exist()
```

Comment:
I don't think `SQLStoreBase.__init__()` is the right place for this. The store is instantiated from a lot of places, including unit tests and `_BatchWindow.__init__()`.

Surely there's a specific point in the flow to do this and not have to repeat it.

Reply:
@gtopper these sub-inits don't create a true "resource", they simply populate the current object with the relevant values.
@davesh0812 I would even remove these sub-inits and move them to the main `__init__`.

Reply:
It does create's the tables @jond01

Reply:
Right, my mistake - I thought only about `self._init_tables()`.

The tables creation should move to `deploy_monitoring_functions` as a part of `enable_model_monitoring`.

Reply:
it is not good enough, the user can have error while trying to run `list_mep`, right after he set the credentials and before he enables MM.

I will put it in the end of `set_cred` in order to avoid this error and also to have some check if the cred are valid.